File: aero523_proj2/rep/plotmesh.py
import numpy as np
import matplotlib.pyplot as plt
from readgri import readgri

#-----------------------------------------------------------
def plotmesh(Mesh, fname):
    V = Mesh['V']; E = Mesh['E']; BE = Mesh['BE']

    f = plt.figure(figsize=(12,12))
    plt.triplot(V[:,0], V[:,1], E, 'k-')
    for i in range(BE.shape[0]):
        plt.plot(V[BE[i,0:2],0],V[BE[i,0:2],1], '-', linewidth=2, color='black')
    plt.axis('equal'); plt.axis('off')
    f.tight_layout(); 
    plt.savefig(fname, bbox_inches='tight')
    plt.close()

File: aero523_proj2/rep/plotmesh.m
function plotmesh

Mesh = readgri('mesh0.gri');

figure(1); clf;
for e = 1:Mesh.nElem,
  I = Mesh.Elem(e,:); I = [I,I(1)];
  plot(Mesh.Node(I,1), Mesh.Node(I,2), 'k-'); hold on;
end
axis equal; axis off
